### PR TITLE
Another batch of Rails 6 changes

### PR DESCRIPTION
**Pull request by dr-itz**
#980

### CODE REVIEW

File: src/java/arjdbc/mysql/MySQLRubyJdbcConnection.java [hunk 1]
```diff
@@ -247,7 +247,9 @@ protected Connection newConnection() throws RaiseException, SQLException {
             connection = super.newConnection();
         }
         catch (SQLException ex) {
-            if (ex.getErrorCode() == 1049) throw newNoDatabaseError(ex);
+            int errorCode = ex.getErrorCode();
+            // access denied, no database
+            if (errorCode == 1044 || errorCode == 1049) throw newNoDatabaseError(ex);
```
Comment (kares) [inline src/java/arjdbc/mysql/MySQLRubyJdbcConnection.java]: access denied as well, hmm AR really keeps being opinionated :wink:

File: src/java/arjdbc/postgresql/PostgreSQLRubyJdbcConnection.java [hunk 2]
```diff
@@ -306,6 +307,24 @@ protected void setArrayParameter(final ThreadContext context,
...
 
+    protected void setDecimalParameter(final ThreadContext context,
+                                       final Connection connection, final PreparedStatement statement,
+                                       final int index, final IRubyObject value,
+                                       final IRubyObject attribute, final int type) throws SQLException {
+        if (value instanceof RubyBigDecimal) {
+            RubyBigDecimal bigDecimal = (RubyBigDecimal) value;
+
+            // too bad RubyBigDecimal.isNaN() isn't public
+            if (bigDecimal.nan_p(context) == context.runtime.getTrue()) {
```
Comment (kares) [inline src/java/arjdbc/postgresql/PostgreSQLRubyJdbcConnection.java]: feel free to use the `context.tru`, `fals`, `nil` "shortcuts"

**Comment by enebo** [comment]
@dr-itz if you want to use the short cuts @kares suggests then feel free to just include them in your next work.

File: src/java/arjdbc/mysql/MySQLRubyJdbcConnection.java (hunk 1 above)
Comment (dr-itz) [reply src/java/arjdbc/mysql/MySQLRubyJdbcConnection.java]: yeah, but the thing is, the MySQL error is kinda wrong. What is tested and I tested manually was a missing database...and you get "access denied"...maybe if you try to access a non-existing DB with root access you get the "no database" error...dunno...

File: src/java/arjdbc/postgresql/PostgreSQLRubyJdbcConnection.java (hunk 2 above)
Comment (dr-itz) [reply src/java/arjdbc/postgresql/PostgreSQLRubyJdbcConnection.java]: ok, will do in the next round

**Pull request merged**